scripts/HealthCheck.java

```diff
@@ -0,0 +1,28 @@
+package uk.gov.digital.ho.hocs;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
+/**
+ * Class used for container health check by docker compose
+ */
+public class HealthCheck {
+
+    public static void main(String[] args) throws IOException, InterruptedException {
+        var client = HttpClient.newHttpClient();
+        var request = HttpRequest.newBuilder()
+                .uri(URI.create("http://localhost:8080/actuator/health"))
```

Comment:
We pass through a `server.port` through application properties, could that be used here? https://github.com/UKHomeOffice/hocs-workflow/blob/8c15c770c1dc710c58c951495ed26181c7630071/src/main/resources/application.properties#L5

Reply:
Thats the external port. I think we always use 8080 inside the docker-compose context. We also have it hardcode in the previous docker-compose.
Having it as a arg to the HealthCheck would make it more reusable......if it wasn't duplicated anyway!

Reply:
I'd argue that we should be defensive with this, we have the property there, there is a chance it could change and it wouldn't be too much to do 🙂

Reply:
I think the problem with using the application.properties is that 8091 isn't the port we want to health check on, 8080 is.

Reply:
The port we want to listen on is the one specified through that `server.port`. We override that in the docker-compose/ kube-deployment to 8080 (but this can be changed to anything).

Reply:
I'm happy to be overruled, just wanted to raise it as a potential improvement 🙂

Reply:
HealthCheck.java now takes the address to check as a command line argument so we can customise in the docker compose file as needed

Reply:
👍🏻